src/tracing/writer.rs

```diff
@@ -409,6 +415,47 @@ impl<W: Write> TraceWriter<W> {
         }
         LOG_STYLE
     }
+
+    fn write_storage_changes(&mut self, node: &CallTraceNode) -> io::Result<()> {
+        let mut changes_map = HashMap::new();
+
+        // For each call trace, compact the results so we do not write the intermediate storage
+        // writes
+        for step in &node.trace.steps {
+            if let Some(ref change) = step.storage_change {
+                let entry =
+                    changes_map.entry((&step.contract, &change.key)).or_insert((None, None));
```

Comment:
node's steps can only contain steps executed in the called contract (`node.trace.address`), so there's no need to index by contract address, and no need to print it along with storage changes